app/src/lib/diff-parser.ts

```diff
@@ -438,6 +461,7 @@ export class DiffParser {
         hunks,
         isBinary: headerInfo.isBinary,
         maxLineNumber: getLargestLineNumber(hunks),
+        hasHiddenBidiChars,
```

Comment:
Couldn't we just run the regex here instead?

```suggestion
        hasHiddenBidiChars: HiddenBidiCharsRegex.test(text),
```